equinix/data_source_metal_device.go

```diff
@@ -197,12 +198,17 @@ func dataSourceMetalDevice() *schema.Resource {
 					},
 				},
 			},
+			"sos_hostname": {
```

Comment:
The device resource will gain the `sos_hostname` attribute in a separate PR to prevent this one from getting even larger than it already is.

Reply:
As an alternative, I've updated this PR to convert the `Read` pathway for the device resource to use `metal-go`.  This enables the resource and the data source to gain the SOS hostname at the same time.  The create/update/delete pathways for the device resource can be converted later in a separate PR.

Reply:
Seems ok to me.